storage-model/src/test/java/no/unit/nva/publication/storage/model/daos/WithCristinIdentifierTest.java

```diff
@@ -0,0 +1,83 @@
+import static no.unit.nva.publication.storage.model.DatabaseConstants.RESOURCES_TABLE_NAME;
+import static no.unit.nva.publication.storage.model.DatabaseConstants.RESOURCE_BY_CRISTIN_ID_INDEX_NAME;
+import static no.unit.nva.publication.storage.model.daos.DaoUtils.sampleResourceDao;
+import static no.unit.nva.publication.storage.model.daos.DaoUtils.toPutItemRequest;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+public class WithCristinIdentifierTest extends ResourcesDynamoDbLocalTest {
+
+    public void init() {
+        super.init();
+    }
+
+    @ParameterizedTest
+    @MethodSource("withPrimaryKeyInstancesProvider")
+    public void dynamoDbClientReturnsItemWithMatchingCristinId(List<ResourceDao> daos) {
+        init();
+        List<WithCristinIdentifier> results = new ArrayList<>();
+        daos.forEach(dao -> client.putItem(toPutItemRequest(dao)));
+        daos.forEach(dao -> results.add(queryDB(dao.getResourceByCristinIdPartitionKey(), dao)));
+        results.forEach(System.out::println);
+        daos.forEach(System.out::println);
+        var expectedItems = daos.toArray(Object[]::new);
+        assertThat(results, containsInAnyOrder(expectedItems));
+    }
+
+    private static Stream<List<ResourceDao>> withPrimaryKeyInstancesProvider()
+            throws InvalidIssnException, MalformedURLException {
+        List<ResourceDao> resources = sampleResourceDaoList(3);
+        return Stream.of(resources);
+    }
+
+    private static List<ResourceDao> sampleResourceDaoList(int sizeOfList)
+            throws MalformedURLException, InvalidIssnException {
+        ArrayList<ResourceDao> resources = new ArrayList<ResourceDao>();
+        for (int i = 0; i < sizeOfList; i++) {
+            resources.add(sampleResourceDao());
+        }
+        return resources;
+    }
+
+    private QueryRequest createQuery(String cristinId) {
+        return new QueryRequest()
+                .withTableName(RESOURCES_TABLE_NAME)
+                .withIndexName(RESOURCE_BY_CRISTIN_ID_INDEX_NAME)
+                .withKeyConditions(createConditionsWithCristinId(cristinId));
+    }
+
+    private WithCristinIdentifier queryDB(String cristinId, WithCristinIdentifier dao) {
+        QueryRequest queryRequest = createQuery(cristinId);
+        return client.query(queryRequest)
+                .getItems()
+                .stream()
+                .map(item -> DynamoEntry.parseAttributeValuesMap(item, dao.getClass()))
+                .collect(SingletonCollector.collectOrElse(null));
+    }
+
+    private Map<String, Condition> createConditionsWithCristinId(String cristinId) {
+        Condition condition = new Condition()
+                .withComparisonOperator(ComparisonOperator.EQ)
+                .withAttributeValueList(new AttributeValue(cristinId));
+        return Map.of(RESOURCES_BY_CRISTIN_ID_INDEX_PARTITION_KEY_NAME, condition);
+    }
+}
```

Comment:
These methods should be in the new interface. Now half of the logic for querying the data lies outside the interface. 
The idea is that we should be able to create a query object (or use an existing object) and use the object to query client.  Also since you wrote the logic, why not put it in the class so that we can use it in the future?

Example (in case we only know the Cristin Identifier)
```
Resource queryResource = Resource.builder()
                .withAdditionalIdentifiers(Set.of(new AdditionalIdentifier("Cristin",randomString())))
                .build();
        ResourceDao query = new ResourceDao(queryResource);
        client.query(query.getByCristinId())
```